specification/network/resource-manager/Microsoft.Network/stable/2021-05-01/firewallPolicy.json

```diff
@@ -1464,6 +1714,236 @@
+      "properties": {
+        "field": {
+          "type": "string",
+          "description": "The name of the field we would like to filter"
+        },
+        "values": {
+          "type": "array",
+          "description": "List of values to filter the current field by",
+          "items": {
+            "type": "string",
+            "description": "Value of the field to filter by"
+          }
+        }
+      }
+    },
+    "OrderBy": {
+      "description": "Describes a column to sort",
+      "type": "object",
+      "properties": {
+        "field": {
+          "type": "string",
+          "description": "Describes the actual column name to sort by"
+        },
+        "asc": {
+          "type": "boolean",
+          "description": "Describes if results should be in ascending order"
+        }
+      }
+    },
+    "QueryResults": {
+      "description": "Query result",
+      "type": "object",
+      "properties": {
+        "allRecords": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Number of total records matching the query."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "When present, the value can be passed to a subsequent query call (together with the same query and scopes used in the current request) to retrieve the next page of data."
+        },
+        "signatures": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SingleQueryResult"
+          },
+          "description": "Array containing the results of the query"
+        }
+      }
+    },
+    "SingleQueryResult": {
+      "type": "object",
+      "properties": {
+        "signatureId": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The ID of the signature"
+        },
+        "mode": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The current mode enforced, 0 - Disabled, 1 - Alert, 2 -Deny",
+          "enum": [
+            0,
+            1,
+            2
+          ]
+        },
+        "severity": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Describes the severity of signature: 1 - Low, 2 - Medium, 3 - High",
+          "enum": [
+            1,
+            2,
+            3
+          ]
+        },
+        "direction": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Describes in which direction signature is being enforced: 0 - Inbound, 1 - OutBound, 2 - Bidirectional",
+          "enum": [
+            0,
+            1,
+            2
+          ]
+        },
+        "group": {
+          "type": "string",
+          "description": "Describes the groups the signature belongs to"
+        },
+        "description": {
+          "type": "string",
+          "description": "Describes what is the signature enforces"
+        },
+        "protocol": {
```

Comment:
>protocol [](http://example.com/codeflow?start=9&length=8)

Are there a fixed set of these that could be modelled as a string enum?

Reply:
No, this information comes from a 3rd party vendor and is parsed in BE